components/constellation/constellation.rs

```diff
@@ -359,6 +360,30 @@ enum ExitPipelineMode {
     Force,
 }
 
+/// A script channel, that closes the script thread down when it is dropped
+pub struct ScriptChan {
+    chan: IpcSender<ConstellationControlMsg>,
+    dont_send_or_sync: PhantomData<Rc<()>>,
+}
```

Comment:
This seems quite dangerous to me. Could we not rely on `Drop`? What happens if that value is forgotten?